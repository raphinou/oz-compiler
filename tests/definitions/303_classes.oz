% Attribute access, including with cell values + accessing non-attribute cells in a method.
local
   C I
in
   class C
      attr
         greeting:'Hello'
         closing:'Bye'
         cellattr:{NewCell 1}
      meth init()
         skip
         %greeting:='Hello'
      end
      meth hello(name:Name<='Doe' firstname:Firstname)
         {Show Name#Firstname}
      end
      meth greet(Name<='Unknown')
         C
      in
         C={NewCell 0}
         {Show @greeting#Name}
         {Show @C}
         {Show @@cellattr}
      end
   end
   I = {New C init()}
   {I hello(firstname:'Alan' name:'Turing')}
   {I hello(firstname:'Jon')}
   {I greet('Mozart')}
   {I greet()}
end
