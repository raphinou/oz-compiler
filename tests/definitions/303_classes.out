'Turing'#'Alan'
'Doe'#'Jon'
'Hello'#'Mozart'
0
1
'Hello'#'Unknown'
0
1
